docs/guides/migration.md

```diff
@@ -0,0 +1,315 @@
+    ```hcl
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+    }
+
+    resource "aci_epg_to_domain" "terraform_epg_to_domain" {
+        application_epg_dn = aci_application_epg.terraform_epg.id
+        tdn                = "uni/phys-phys"
+    }
+    ```
+
+    ***New Configuration***
+    ```
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+        relation_to_domains = [
+            {
+                target_dn = "uni/phys-phys"
+            }
+        ]
+    }
+    ```
+
+    !> Child resources should not be used in combination with nested attributes in parent resources. Doing so will result in unexpected behaviour.
+
+    In this scenario the child configuration should not be destroyed, but should be unmanaged (removed from state) before executing the plan. Terraform will display destroy intent in the `terraform plan` output when executed without unmanaging the child resource first.
+
+    ```bash
+    Terraform will perform the following actions:                            
+
+    # aci_epg_to_domain.terraform_epg_to_domain will be destroyed
+    # (because aci_epg_to_domain.terraform_epg_to_domain is not in configuration)
+    - resource "aci_epg_to_domain" "terraform_epg_to_domain" {
+      - application_epg_dn    = "uni/tn-terraform_tenant/ap-terraform_ap/epg-terraform_epg" -> null
+      - tdn                   = "uni/phys-phys" -> null
+      ...
+      - vmm_id                = jsonencode({})
+        # (7 unchanged attributes hidden)
+    }
+
+    Plan: 0 to add, 0 to change, 1 to destroy.
+    ```
+
+    The [terraform state rm](https://developer.hashicorp.com/terraform/cli/commands/state/rm) command `terraform state rm aci_epg_to_domain.terraform_epg_to_domain` should be used to unmanage without destroying a resource. The output should reflect the succesful removal of the resource from state.
+
+    ```bash
+    Removed aci_epg_to_domain.terraform_epg_to_domain
+    Successfully removed 1 resource instance(s).
+    ```
+
+    The `terraform plan` output should reflect no changes, when the resource is unmanaged.
+    
+    ```bash
+    No changes. Your infrastructure matches the configuration.
+
+    Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
+    ```
+
+    The state file does not reflect the changes yet, because a refresh has not taken place. This can be done with the `terraform refresh` or `terraform apply -refresh-only` commands, which do not modify the objects in apic, but only modifies the state file. For more information see [refresh](https://developer.hashicorp.com/terraform/cli/commands/refresh) and [planning-modes](https://developer.hashicorp.com/terraform/cli/commands/plan#planning-modes) documentation of Terraform.
```

Comment:
```suggestion
    The state file does not reflect the changes yet, because a refresh has not taken place. This can be done with the `terraform refresh` or `terraform apply -refresh-only` commands, which do not modify the objects in APIC, but only modifies the state file. For more information see [refresh](https://developer.hashicorp.com/terraform/cli/commands/refresh) and [planning-modes](https://developer.hashicorp.com/terraform/cli/commands/plan#planning-modes) documentation of Terraform.
```